### PR TITLE
chore(*): migrate to golangci-lint

### LINKED ISSUE #287: migrate to golangci-lint
[golangci-lint](https://github.com/golangci/golangci-lint) is faster than [gometalinter](https://github.com/alecthomas/gometalinter), easier to use, and rich in configuration. And since `gometalinter` is DEPRECATED for now, we should migrate from `gometalinter` to `golangci-lint`

related: https://github.com/caicloud/nirvana/issues/31

/cc @ddysher @kdada

### PR DESCRIPTION
**Pull request by iawia002**
**What this PR does / why we need it**:

close #287 

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes #

Reference to #

**Special notes for your reviewer**:

/cc @your-reviewer

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: service/context.go [hunk 1]
```diff
@@ -206,6 +206,8 @@ func (c *response) Flush() {
 }
 
 // CloseNotify is a disguise of http.response.CloseNotify().
+//
+// Deprecated: use `http.Request.Context.Done()` as instead.
 func (c *response) CloseNotify() <-chan bool {
```
Comment (kdada) [inline service/context.go]: The implementation is not correct. This method makes users have no way to check if the response supports http.CloseNotifier. Do you have time to fix this in a new PR?

**Comment by kdada** [comment]
LGTM for the rest.

File: service/context.go (hunk 1 above)
Comment (iawia002) [reply service/context.go]: So I deleted it directly in this PR? As mentioned in #288

Comment (kdada) [reply service/context.go]: I mean that the response may not implement `http.CloseNotifier`.

The method should be separated into two parts:
```
type response struct {...}


type responseWithCloseNotify struct {
    *response
}

func (c *responseWithCloseNotify) CloseNotify() <-chan bool {...}
```

Then wrap response with the new struct.

```

rw = &response{}

if _, ok:=rw.(http.CloseNotifier); ok {
    rw = &responseWithCloseNotify{rw}
}
```

If the method is removed from the go library in the future, the `response` still works.

Comment (iawia002) [reply service/context.go]: You mean still keep this function? As you replied `Just do it.` in #288 I thought we can just delete this function.

Comment (kdada) [reply service/context.go]: It seems that we need to retain it because of the behavior of go library.

Comment (iawia002) [reply service/context.go]: Fine, I will fix this in a new PR

**Pull request merged**